### ISSUE TITLE
[bug] Attachment fetching: body size too large

### ISSUE THREAD

**Issue by firescry** [superseriousbusiness/gotosocial#2308]
### Describe the bug with a clear and concise description of what the bug is.

I follow an account which frequently posts a lot of video attachments. I've noticed that most of the attachments are never fetched to my GoToSocial instance.

Example of a post: https://snabelen.no/@Mediablikk/111316607666931868

In the logs I can see the following error:
```
Oct 29 06:17:16 xyz gotosocial[922]: func=workers.(*Processor).ProcessFromFediAPI level=INFO activityType=Create objectType=Note toAccount=canary requestID=sdqdtycb040010ampnp0 msg="processing from fedi API"
Oct 29 06:17:16 xyz gotosocial[922]: func=server.glob..func1.Logger.func13.1 level=INFO latency="12.466465ms" userAgent="http.rb/5.1.1 (Mastodon/4.2.1; +https://snabelen.no/)" method=POST statusCode=202 path=/users/canary/inbox clientIP=57.128.95.181 requestID=sdqdtycb040010ampnp0 msg="Accepted: wrote 45B"
Oct 29 06:17:16 xyz gotosocial[922]: func=httpclient.(*Client).DoSigned level=INFO method=GET url=https://cdn.masto.host/snabelenno/media_attachments/files/111/316/606/768/210/211/original/595fb2b8ed300717.mp4 requestID=sdqdtycb040010ampnp0 pubKeyID=https://social.firescry.com/users/canary/main-key msg="performing request"
Oct 29 06:17:19 xyz gotosocial[922]: func=dereferencing.(*deref).fetchStatusAttachments level=ERROR requestID=sdqdtycb040010ampnp0 msg="error loading attachment: store: error executing data function: body size too large"
Oct 29 06:17:19 xyz gotosocial[922]: func=httpclient.(*Client).DoSigned level=INFO method=GET url=https://snabelen.no/users/Mediablikk/statuses/111316607666931868/replies?only_other_accounts=true&page=true pubKeyID=https://social.firescry.com/users/canary/main-key msg="performing request"
Oct 29 06:17:20 xyz gotosocial[922]: func=server.glob..func1.Logger.func13.1 level=INFO latency="1.839637ms" userAgent="http.rb/5.1.1 (Mastodon/4.2.1; +https://snabelen.no/)" method=GET statusCode=200 path=/users/canary/main-key clientIP=57.128.95.181 requestID=11ydtycb04000mknsc90 msg="OK: wrote 561B"
Oct 29 06:17:20 xyz gotosocial[922]: func=server.glob..func1.Logger.func13.1 level=INFO latency="2.439134ms" userAgent="http.rb/5.1.1 (Mastodon/4.2.1; +https://snabelen.no/)" method=GET statusCode=200 path=/users/canary clientIP=57.128.95.181 requestID=knydtycb04001kn6672g msg="OK: wrote 1.17kiB"
Oct 29 06:17:20 xyz gotosocial[922]: func=server.glob..func1.Logger.func13.1 level=INFO latency="3.2415ms" userAgent="http.rb/5.1.1 (Mastodon/4.2.1; +https://snabelen.no/)" method=GET statusCode=200 path=/users/canary clientIP=57.128.95.181 requestID=61yxtycb040010r7m910 msg="OK: wrote 1.17kiB"
```

Video attachment was rejected due to its size being too large.

### What's your GoToSocial Version?

v0.12.1 (git-5fdc005)

### GoToSocial Arch

arm64

### What happened?

There is a hard-coded 40MB limit for attachments - [superseriousbusiness/gotosocial/blob/v0.12.1/internal/httpclient/client.go#L138
](https://github.com/superseriousbusiness/gotosocial/blob/5fdc005061c773fda6e0da32abcc6f79ddee80ed/internal/httpclient/client.go#L138)
Rejected video is more than 50MB.

### What you expected to happen?

I'd like that the video was fetched to my instance :)
I think the limit could be parameterized, so admins could decide maximum size for attachments to be downloaded.

### How to reproduce it?

Fetch a post with big attachment (e.g. https://snabelen.no/@Mediablikk/111316607666931868). Post will be fetched without attachment.

### Anything else we need to know?

Similar topic of too big media attachments was also mentioned in #1013 (but with slightly different scope and proposal for resolution) and in already closed (but not merged) #810.

I'm not sure what would be preferred way to handle this issue - fetch big attachments with parameterized maximum size or redirect to original instance to show them. For the first approach I think I could work on providing a patch. The other could be to difficult for me though :)

**Comment by tsmethurst** [comment]
I'll chat about it with the other devs and see what they think. Probably it's better to do *both* things, ie., allow something like 'media-remote-max-size' to be set, and also display a placeholder in case a piece of media is too large / not available.

**Comment by firescry** [comment]
Have you had a chance to discuss the idea to parametrise the max size of remote media files? Would that be a preferred way to go?
I'll have some spare time in coming weeks so I could propose a patch implementing it 😊

**Comment by Jasdemi** [comment]
Can this be closed as the max remote media size can be changed with [`media-remote-max-size` ](https://docs.gotosocial.org/en/latest/configuration/media/#__codelineno-0-22)now?

**Comment by firescry** [comment]
Thanks for notice, @Jasdemi. This issue was indeed recently resolved in #3090.

Thank you all!
